daemon/src/order/maker.rs

```diff
@@ -0,0 +1,385 @@
+    n_payouts: usize,
+    decision_senders: HashMap<OrderId, oneshot::Sender<protocol::Decision>>,
+    db: sqlite_db::Connection,
+    latest_offers: MessageChannel<xtra_libp2p_offer::maker::GetLatestOffers, Option<MakerOffers>>,
+}
+
+impl Actor {
+    pub fn new(
+        n_payouts: usize,
+        oracle_pk: XOnlyPublicKey,
+        get_announcement: MessageChannel<
+            oracle::GetAnnouncement,
+            Result<olivia::Announcement, NoAnnouncement>,
+        >,
+        (db, process_manager): (sqlite_db::Connection, xtra::Address<process_manager::Actor>),
+        (build_party_params, sign): (
+            MessageChannel<wallet::BuildPartyParams, Result<PartyParams>>,
+            MessageChannel<wallet::Sign, Result<PartiallySignedTransaction>>,
+        ),
+        projection: xtra::Address<projection::Actor>,
+        latest_offers: MessageChannel<
+            xtra_libp2p_offer::maker::GetLatestOffers,
+            Option<MakerOffers>,
+        >,
+    ) -> Self {
+        Self {
+            executor: command::Executor::new(db.clone(), process_manager),
+            oracle_pk,
+            get_announcement,
+            build_party_params,
+            sign,
+            projection,
+            n_payouts,
+            decision_senders: HashMap::default(),
+            db,
+            latest_offers,
+        }
+    }
+
+    async fn await_order(
+        &mut self,
+        framed: &mut Framed<Substream, JsonCodec<MakerMessage, TakerMessage>>,
+    ) -> Result<TakerMessage> {
+        let order = framed
+            .next()
+            .timeout(ORDER_TIMEOUT, || tracing::debug_span!("receive order"))
+            .await
+            .context("Timout when waiting for order")?
+            .context("Stream terminated")?
+            .context("Unable to decode order")?;
+
+        Ok(order)
+    }
+
+    async fn pick_offer(&self, offer_id: OfferId) -> Result<Offer> {
+        // TODO: Evaluate if this can be a problem during sending new offers to all connected peers
+        // -> Should be fine because the offer is already outdated by then and we should actually
+        // wait until finished?
+        // -> Cannot use send_async_next here - is that expected?
+        let latest_offers = match self
+            .latest_offers
+            .send(xtra_libp2p_offer::maker::GetLatestOffers)
+            .await
+        {
+            Ok(latest_offers) => latest_offers,
+            Err(e) => {
+                bail!("Failed to retrieve latest offer from offers actor: {e:#}")
+            }
+        };
+
+        let offer = match latest_offers {
+            None => bail!("No latest offers available"),
+            Some(offers) => offers.pick_offer_to_take(offer_id),
+        };
+
+        match offer {
+            None => bail!("Offer with id {offer_id} not found in current offers"),
+            Some(offer) => Ok(offer),
+        }
+    }
+}
+
+#[xtra_productivity(message_impl = false)]
```

Comment:
`message_impl` that used to implement message handlers is not needed anymore. :)

```suggestion
#[xtra_productivity]
```